packages/babel-plugin-transform-inline-consecutive-adds/__tests__/transform-inline-consecutive-adds-test.js

```diff
@@ -0,0 +1,225 @@
+    const source = unpad(`
+      const foo = {};
+      foo.a = function () {
+        console.log(3);
+      };
+      foo.b = foo.a();
+    `);
+    const expected = unpad(`
+      const foo = {
+        a: function () {
+          console.log(3);
+        }
+      };
+
+      foo.b = foo.a();
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should not collapse if has indirect dependency issues", () => {
+    const source = unpad(`
+      var foo = {};
+      foo.a = 4;
+      foo.b = cat();
+      function cat() {
+        return bar();
+      }
+      function bar() {
+        console.log(foo);
+        return 0;
+      }
+    `);
+    const expected = unpad(`
+      var foo = {
+        a: 4
+      };
+
+      foo.b = cat();
+      function cat() {
+        return bar();
+      }
+      function bar() {
+        console.log(foo);
+        return 0;
+      }
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+
+  it("should not collapse if has indirect, nested dependency issues", () => {
+    const source = unpad(`
+      var foo = {};
+      foo.a = 4;
+      foo.b = cat();
+      function cat() {
+        return bar();
+        function bar() {
+          console.log(foo);
+          return 0;
+        }
+      }
+    `);
+    const expected = unpad(`
+      var foo = {
+        a: 4
+      };
+
+      foo.b = cat();
+      function cat() {
+        return bar();
+        function bar() {
+          console.log(foo);
+          return 0;
+        }
+      }
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should collapse normal computed properties", () => {
+    const source = unpad(`
+      var foo = {};
+      foo["a"] = 0;
+      foo[4] = 1;
+    `);
+    const expected = unpad(`
+      var foo = {
+        "a": 0,
+        4: 1
+      };
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should not collapse computed properties with dependency issues", () => {
+    const source = unpad(`
+      var foo = {};
+      foo[bar()] = 0;
+      function bar() {
+        console.log(foo);
+        return 0;
+      }
+    `);
+    expect(transform(source)).toBe(source);
+  });
+
+  it("should collapse statements with multiple assignments", () => {
+    const source = unpad(`
+      var foo = {};
+      foo.a = 0, foo.b = 2;
+    `);
+    const expected = unpad(`
+      var foo = {
+        a: 0,
+        b: 2
+      };
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should not collapse statements with multiple assignments and dependency issues", () => {
+    const source = unpad(`
+      var foo = {};
+      foo.a = 0, foo.b = bar();
+      function bar() {
+        console.log(foo);
+        return 0;
+      }
+    `);
+    expect(transform(source)).toBe(source);
+  });
+
+  it("should collapse statements for arrays", () => {
+    const source = unpad(`
+      var foo = [1, 2];
+      foo.push(3, 4), foo.push(5);
```

Comment:
I wonder if we should bother with property-based assignments:

```js
var foo = [];
foo[1] = 'blah';
foo[2] = 'blah';
```
transforming to:
```
var foo=[,"blah","blah"];
```

Closure Compiler does this, btw — https://closure-compiler.appspot.com/home#code%3Dvar%2520foo%2520%253D%2520%255B%255D%253B%250Afoo%255B1%255D%2520%253D%2520'blah'%253B%250Afoo%255B2%255D%2520%253D%2520'blah'%253B

They also use smart heuristic (based on final length?) and transform this:

```js
var foo = [];
foo[3] = 'blah';
foo[5] = 'blah';
foo[7] = 'blah';
```
to:
```js
var foo=[,,,"blah",,"blah",,"blah"];
```
but not:
```js
var foo = [];
foo[10] = 'blah';
```